Retain Previous Selections and Display New Menu Without Overwriting

FLASH2332 opened the pull request:
Changes implemented :
- "Back" removes the last selection.
- "Go Back to Main Menu" clears history.
- Selections are properly appended.

The selections are displayed on top of each new menu

![image](https://github.com/user-attachments/assets/d75b8bb4-edcb-4f15-99ae-f7c526e87107)

FLASH2332 committed:
Retain the previously selected options while updating the menu

FLASH2332 committed:
Merge branch 'CSE-25:main' into main

Ashrockzzz2003 committed:
fmt go code.

Ashrockzzz2003 reviewed (changes requested):
Looks clean!

For this case
`Btech -> 7th Sem -> 2023 ->`
In the final step where the list of question papers are displayed, the preview is not there. Can you add it there too ?

Ashrockzzz2003 reviewed (commented):
Try solving it, file a bug, we'll open it up to the community

FLASH2332 committed:
Improved visibility of options and path-selection in final menu

FLASH2332 committed:
Merge branch 'main' of https://github.com/FLASH2332/amrita_pyq

FLASH2332 commented:
Corrected it
![image](https://github.com/user-attachments/assets/14bd378b-2b97-4aba-bd04-a2efdcebe980)
![image](https://github.com/user-attachments/assets/400b8481-671c-4806-ac10-c5b69eae61f8)

Currently only 10 units of height are shown should I increase it ?
Is this fine also I found the Exiting.. text is very plain shall I make it as a spinner ?

Ashrockzzz2003 reviewed (commented):
Awesome!!
Sure go ahead @FLASH2332

FLASH2332 committed:
Updated the height of final menu and modified exit message as spinner

FLASH2332 commented:
Changes have been made

Ashrockzzz2003 committed:
Limit height only when number of qn papers > 10.

Ashrockzzz2003 reviewed (changes requested):
Can you make exit to be a module and use it in all screens

semChoose
year
root ... etc

Right now it's only in year.
All places.

FLASH2332 committed:
Add reusable QuitWithSpinner function
Added quit option to all the menus(root,semtable,semchoose,year)
Implemented function is being called for quit option across all menus

FLASH2332 commented:
Made the changes and verified its working in all menus

Ashrockzzz2003 committed:
Run go fmt.

Ashrockzzz2003 reviewed (approved):
LGTM
Amazing work Jayadev!
Thanks for the contribution.

Merging this one now!

Ashrockzzz2003 commented:
/bounty 50 @FLASH2332
